### PR TITLE
[MKISOFS] Fix Clang check for macOS platforms

### PR DESCRIPTION
**Pull request by wjk**
## Purpose

On macOS, CMAKE_C_COMPILER_ID is "AppleClang". While this certainly is Clang, it does not match the exact string "Clang" that is being checked for, and as a result the warning flags guarded thereby are not passed to the compiler.

JIRA issue: none

## Proposed changes

Get CMake to recognize both "Clang" and "AppleClang".

### CODE REVIEW

**Comment by wjk** [comment]
I have no idea what this error means:

```
2024-06-19T19:27:40.9701971Z D:\a\reactos\reactos\src\ntoskrnl\io\iomgr\symlink.c(119): fatal error C1090: PDB API call failed, error code '23': (0x000006BA)
```

This is what has caused that one check to fail. I suspect that this is a transient error, but I have no way of forcing that job to rerun. Any pointers would be appreciated.

**Comment by SergeGautherie** [comment]
> CMAKE_C_COMPILER_ID is "AppleClang"

There are other similar occurrences.
Is MkIsoFs really the only occurrence that "fails"?

> transient error

It is...

**Comment by wjk** [comment]
> There are other similar occurrences.

All of the other uses I can find (using `ack 'STREQUAL "Clang"'`) are in ReactOS core code, which should be compiled by some variety of RosBE. This use (and only this use) was in the host-tools, which is not compiled using RosBE. Should I change the others?

**Comment by SergeGautherie** [comment]
> ReactOS core code, which should be compiled by some variety of RosBE.

Ok, in that very case.

> host-tools, which is not compiled using RosBE.

I suggest to add an explicit code comment, to explain why `MATCHES` is used here.

> Should I change the others?

No, if unneeded for the time being.

File: sdk/tools/mkisofs/CMakeLists.txt [hunk 1]
```diff
@@ -107,7 +107,8 @@ else()
     # Silence compilers checking for invalid formatting sequences.
     target_compile_options(libschily PRIVATE "-Wno-format")
 
-    if(CMAKE_C_COMPILER_ID STREQUAL "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "15")
+    # MATCHES must be used here because on macOS, CMake uses the compiler ID "AppleClang".
+    if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "15")
```
Comment (SergeGautherie) [inline sdk/tools/mkisofs/CMakeLists.txt:111]: A bit stricter:
```suggestion
    if(CMAKE_C_COMPILER_ID MATCHES "Clang$" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "15")
```

**Comment by HBelusca** [comment]
Agreed with Serge: the change is because this is the only usage of "Clang" cmake check in a _**host-tool**_, for which the native platform compiler is used.

**Comment by wjk** [comment]
Any reason why this PR has not been merged? From my perspective it seems complete.

**Review by tkreuzer — APPROVED** [review]
Looks ok to me. I have no idea about MAC though.

**Review by DarkFire01 — APPROVED** [review]
WorksOnMyMachine

**Pull request merged**